SparkPost/2web2ui
# FE-923 - Adds Hibana mobile navigation

[FE-923](https://jira.int.messagesystems.com/browse/FE-923)

### What Changed
- Adds new mobile navigation
- Adds `cypress-plugin-tab` which provides the `cy.tab()` command for keyboard interaction testing
- Adds `react-focus-lock` for managing tab traps within modal-like interfaces

### How To Test
1. Reduce the screen width below 960px
1. Open the navigation using the button in the top right
1. Verify the appropriate links are present and that subsections render as expandable/collapsible sections
1. Verify that all portions of the navigation are accessible using a keyboard only
1. Verify that when the navigation is open, a tab trap is implemented - when using `tab` and `shift + tab` the user cannot tab outside of the navigation - but _only_ when the navigation is open!

### To Do
- [x] Should I break `MobileNavigation.js` in to multiple files? I've been leaning towards larger files with multiple components lately, but that might just be me!
